dask_bigquery/core.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import annotations
+
+import logging
+import warnings
+from collections.abc import Iterable
+from contextlib import contextmanager
+from functools import partial
+
+import dask
+import dask.dataframe as dd
+import pandas as pd
+import pyarrow
+from google.cloud import bigquery, bigquery_storage
+
+
+@contextmanager
+def bigquery_client(project_id="dask-bigquery", with_storage_api=False):
```

Comment:
maybe link to https://github.com/googleapis/google-cloud-python/issues/9457 and/or https://github.com/googleapis/gapic-generator-python/issues/575 for context (no pun intended)

Reply:
also re project_id, you probably ought make it default to `None` and infer from the global context, don't remember off the top of my head where you grab that from

Reply:
> maybe link to [googleapis/google-cloud-python#9457](https://github.com/googleapis/google-cloud-python/issues/9457) and/or [googleapis/gapic-generator-python#575](https://github.com/googleapis/gapic-generator-python/issues/575) for context (no pun intended)

Do you mean add them as a comment?

Reply:
yeah, just bc eventually there will probably be one upstream that could be used but right now there's not